Remotion/ObjectBinding/Web/UI/Controls/Validation/BocListValidationResultDispatchingValidator.cs

```diff
@@ -36,16 +36,71 @@ public void DispatchValidationFailures (IBusinessObjectValidationResult validati
       ArgumentUtility.CheckNotNull("validationResult", validationResult);
 
       var bocListControl = GetControlToValidate();
+      var dispatchToValidationFailureRepository = bocListControl
+          .GetColumnDefinitions()
+          .Any(cd => cd is BocValidationErrorIndicatorColumnDefinition);
 
+      if (dispatchToValidationFailureRepository)
+        DispatchToValidationFailureRepository(validationResult, bocListControl);
+      else
+        DispatchToValidationResultDispatchers(validationResult, bocListControl);
+    }
+
+    private void DispatchToValidationResultDispatchers (IBusinessObjectValidationResult validationResult, BocList bocListControl)
+    {
       var validatorsMatchingToControls = EnumerableUtility.SelectRecursiveDepthFirst(
               bocListControl as Control,
-              child => child.Controls.Cast<Control>().Where(item => !(item is INamingContainer)))
+              child => child.Controls.Cast<Control>().Where(item => item is not INamingContainer))
           .OfType<IBusinessObjectBoundEditableWebControlValidationResultDispatcher>();
 
       foreach (var validator in validatorsMatchingToControls)
         validator.DispatchValidationFailures(validationResult);
     }
 
+    private void DispatchToValidationFailureRepository (IBusinessObjectValidationResult validationResult, BocList bocListControl)
+    {
+      var validationFailureRepository = bocListControl.ValidationFailureRepository;
+      validationFailureRepository.ClearAllValidationFailures();
+
+      var rowObjects = bocListControl.Value?.Cast<IBusinessObject>();
+
+      if (rowObjects != null)
+      {
+        var columnsAndMatchers = bocListControl
+            .GetColumnDefinitions()
+            .OfType<IBocColumnDefinitionWithValidationSupport>()
+            .Select(c => new { Column = (BocColumnDefinition)c, Matcher = c.GetValidationFailureMatcher() })
+            .ToArray();
+
+        foreach (var rowObject in rowObjects)
+        {
+          foreach (var columnAndMatcher in columnsAndMatchers)
+          {
+            var matchedFailures = columnAndMatcher.Matcher.GetMatchingValidationFailures(rowObject, validationResult);
+
+            if (matchedFailures.Count > 0)
+              validationFailureRepository.AddValidationFailuresForDataCell(rowObject, columnAndMatcher.Column, matchedFailures);
+          }
+
+          var otherRowFailures = validationResult.GetValidationFailuresForBusinessObject(rowObject);
+          if (otherRowFailures.Count > 0)
+            validationFailureRepository.AddValidationFailuresForDataRow(rowObject, otherRowFailures);
+        }
+      }
+
+      var businessObject = bocListControl.DataSource?.BusinessObject;
+      var property = bocListControl.Property;
+
+      if (businessObject == null)
+        return;
+      if (property == null)
+        return;
+
+      var listPropertyFailures = validationResult.GetValidationFailures(businessObject, property, true);
```

Comment:
please use a parameter name to better document the code. it's somehting I try to do with any non-obvious boolean these days (modern c#)

Reply:
I think we should discuss this general problem as a team since the IDEs already add the inline tooltip already:

![image](https://github.com/re-motion/Framework/assets/5082673/68d1575a-3d0a-41e2-970e-43d8e55bff0d)

Adding an explicit parameter name then only has the benefit of being easier to read without an IDE (e.g. in Github reviews)

Reply:
Exactly. Also, for those who disable this settng in the IDE because they find it distracting. Having this in case of multiple booleans is incredibly helpful when guarding against copy&paste errors, broken refactorings, and api upgrades. E.g. if there is an extra boolean parameter added in a method and you do not use the automated refactoring (or if you use a library which introduces a new parameter or changes a signature), the information which boolean we referred to is available right there in the code and will fail or do the correct thing. All these little guards go away if you have the IDE supply the information inline. 

I did anwser here because it popped up in my inbox, it's not meant to start a lengthy async discussion.